packages/@aws-cdk/aws-elasticloadbalancing/lib/load-balancer.ts

```diff
@@ -398,6 +399,34 @@ export class LoadBalancer extends Resource implements IConnectable {
       Port.tcp(instancePort),
       `Port ${instancePort} LB to fleet`);
   }
+
+  /**
+   * add instance to the load balancer.
+   * @internal
+   * @param instanceId
+   */
+  public _addInstanceId(instanceId: string) { this.instanceIds.push(instanceId); }
+}
+
+/**
+ * An EC2 instance that is the target for load balancing
+ *
+ */
+export class InstanceTarget implements ILoadBalancerTarget {
+  readonly connections: Connections;
+  /**
+   * Create a new Instance target.
+   *
+   * @param instance Instance to register to.
+   * @param port Override the default port for the target.
+   */
+  constructor(public readonly instance: Instance, public readonly port: number) {
+    this.connections = instance.connections;
+  }
+  public attachToClassicLB(loadBalancer: LoadBalancer): void {
+    loadBalancer._addInstanceId(this.instance.instanceId);
+    this.connections.allowFrom(loadBalancer.connections, Port.tcp(this.port));
```

Comment:
I have one last question about the `port` parameter. You are saying that if `port` is given, it "overrides the default port".

But think about the following:

* **The parameter is not optional**. Normally, if you say that a value "overrides" a different value, you have an option to use the default as well. This parameter is not optional, so while you could say it "overrides" something, that is not really a useful statement: it is just going to be the port on the target.
* Second, the parameter is **only used in the SecurityGroup** (firewall rules). We're not using the port when we are configuring the load balancer itself. That port is configured elsewhere. 

So that means we are in the following situation: the load balancer will forward to instance port A, while we open up the firewall on port B. It is now the user's responsibility to make sure that `A == B`. If they make a mistake, then the load balancer will try to use a port that the security groups don't allow connections on, and the load balancing will not work.

This kind of redundancy, with opportunities for misconfiguration, is exactly what CDK tries to prevent! 

So our ideal solution will be for the `InstanceTarget` to make sure that the user doesn't have to specify the port number again. The load balancer already *knows* what port it's trying to connect to. We should reuse that.